imports/collections/schemas/catalog.js

```diff
@@ -1,20 +1,638 @@
+ * @property {Number} lowInventoryWarningThreshold optional, default value: `0`
+ * @property {Metafield[]} metafields optional
+ * @property {Number} minOrderQuantity optional, default value: `1`
+ * @property {String} optionTitle optional
+ * @property {String} originCountry optional
+ * @property {CatalogPriceRange} price required
+ * @property {ProductPricingInfoByCurrency{}} pricing required
+ * @property {String} shopId required
+ * @property {String} sku optional
+ * @property {String} taxCode optional, default value: `"0000"`
+ * @property {String} taxDescription optional
+ * @property {String} title optional
+ * @property {Date} updatedAt optional
+ * @property {String} variantId required
+ * @property {Number} weight optional, default value: `0`
+ * @property {Number} width optional, default value: `0`
+ */
+export const CatalogProductVariant = new SimpleSchema({
+  "_id": {
+    type: String,
+    label: "Catalog product variant Id"
+  },
+  "barcode": {
+    type: String,
+    label: "Barcode",
+    optional: true
+  },
+  "createdAt": {
+    type: Date,
+    label: "Date/time this variant was created at",
+    optional: true
+  },
+  "height": {
+    type: Number,
+    label: "Height",
+    min: 0,
+    optional: true,
+    defaultValue: 0
+  },
+  "index": {
+    type: SimpleSchema.Integer,
+    label: "The position of this variant among other variants at the same level of the product-variant-option hierarchy"
+  },
+  "inventoryManagement": {
+    type: Boolean,
+    label: "True if inventory management is enabled for this variant"
+  },
+  "inventoryPolicy": {
+    type: Boolean,
+    label: "True if inventory policy is enabled for this variant"
+  },
+  "isLowQuantity": {
+    type: Boolean,
+    label: "Indicates that the product quantity is too low"
+  },
+  "isSoldOut": {
+    type: Boolean,
+    label: "Indicates when the product quantity is zero"
+  },
+  "isTaxable": {
+    type: Boolean,
+    label: "Indicates if a product is taxable",
+    index: 1,
+    defaultValue: false
+  },
+  "length": {
+    type: Number,
+    label: "Length",
+    min: 0,
+    optional: true,
+    defaultValue: 0
+  },
+  "lowInventoryWarningThreshold": {
+    type: SimpleSchema.Integer,
+    label: "Warn at",
+    min: 0,
+    optional: true,
+    defaultValue: 0
+  },
+  "metafields": {
+    type: Array,
+    label: "Metafields",
+    optional: true
+  },
+  "metafields.$": {
+    type: Metafield
+  },
+  "minOrderQuantity": {
+    type: SimpleSchema.Integer,
+    label: "Minimum quantity per item in an order",
+    defaultValue: 1,
+    optional: true
+  },
+  "optionTitle": {
+    type: String,
+    label: "Option title",
+    optional: true
+  },
+  "originCountry": {
+    type: String,
+    label: "Origin country",
+    optional: true
+  },
+  "price": {
+    type: Number
+  },
+  "pricing": {
+    type: Object,
+    label: "Pricing"
+  },
+  "pricing.$": {
+    type: ProductPricingInfoByCurrency
+  },
```

Comment:
This `pricing.$` won't actually work. SimpleSchema doesn't support validating objects with arbitrary keys. You'll have to add `blackbox: true` to `pricing` or the cleaning will probably empty out the `pricing` object when inserting.

And then you can remove the `pricing.$` key.

Reply:
@aldeed Does this mean we don't need the `ProductPricingInfoByCurrency` schema anymore, as this is the only place it's used.

Reply:
Yes, unfortunately there's no way to validate those right now.